### ISSUE TITLE
Support for adding new Backend Address Pool to an existing Application Gateway

### ISSUE THREAD

**Issue by d-lee** [terraform-providers/terraform-provider-azurerm#5225]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

We would like an ability to add new backend pool to an existing application gateway without having to tear the gateway down. It is possible to do this via Azure CLI but Terraform doesn't seem to support such use-case.

**Comment by boillodmanuel** [comment]
I agree, and I would be great to be able to add any part of an application gateway as a separate resource
: 
- `azurerm_application_gateway_frontend_port`
- `azurerm_application_gateway_frontend_ip_configuration`
- `azurerm_application_gateway_backend_address_pool`
- `azurerm_application_gateway_backend_http_settings`
- `azurerm_application_gateway_http_listener_request_routing_rule`
- `azurerm_application_gateway`

This could already be done with `azurerm_loadbalancer` and separate resources :
- `azurerm_lloadbalancer_backend_address_pool`
- `azurerm_loadbalancer_nat_rule`
- `azurerm_loadbalancer_outbound_rule`
- `azurerm_loadbalancer_probe`
- `azurerm_loadbalancer_rule.html.markdown`

**Comment by boillodmanuel** [comment]
After some investigations, I notice theses resources are missing from the azure-sdk-for-go, so I opened 2 issues : 
- azure-sdk-for-go : https://github.com/Azure/azure-sdk-for-go/issues/7234
- azure-rest-api-specs : https://github.com/Azure/azure-rest-api-specs/issues/8252

If anyone could vote for these root issues 👏

**Comment by tombuildsstuff** [comment]
@boillodmanuel both of those are blocked on this issue unfortunately: https://github.com/Azure/azure-rest-api-specs/issues/2313

**Comment by boillodmanuel** [comment]
Indeed @tombuildsstuff . I voted for this issue too. Thanks for sharing

**Comment by boillodmanuel** [comment]
Asked on azure feedback site: https://feedback.azure.com/forums/217313-networking/suggestions/39634588-add-rest-apis-and-sdk-to-manage-application-gatewa
If people following this issue could vote on it, it will be great 👍

**Comment by Dmitry1987** [comment]
Is this for real, there is no option to add a new backend address pool like in AWS ALB (a new target group to route to) when app-gateway is already operational in production??  
Is it a TF module limitation or Azure cloud itself?

**Comment by abhishekjerry2000** [comment]
Hi All,

Can you please update on this? Its really a pain to update the Application gateway in case its existing.

**Comment by Santosh-Kunadi** [comment]
Is terraform considering this request ?

**Comment by Sennar19** [comment]
Hi all! Any news?
